test/unit/test_utils.js

```diff
@@ -15,6 +15,14 @@
 
 import { CMapCompressionType } from '../../src/shared/util';
 
+class NodeFileReader {
+  constructor(params) {
+    var fs = require('fs');
+    var file = fs.readFileSync(params.path);
+    return new Uint8Array(file);
```

Comment:
Even though this worked, you're absolutely right that having a constructor return a value is an anti-pattern!
The easiest solution might have been use a regular utility function, but I figured it'd be nice to have something factory-like. Hence I settled on a class with a *static* method, which means that you don't need to instantiate anything before using it.
With the new code, this is now as simple as calling `NodeFileReaderFactory.fetch({ ... })`; which I hope you'll agree is a better approach here!